auth/auth.js

```diff
@@ -30,10 +31,18 @@ async function generateToken(email, clientdata) {
 }
 
 async function generateUser(userDetails) {
+  await bcrypt.hashSync(userDetails.password, SECRET, (err, hash) => {
+    userDetails.password = hash;
+  });
```

Comment:
1. bcrypt is not promise based. Simply return the hash from callback or, you can also omit returning hash value because, hashSync itself returns the hash password.
2. In place of secret you have to pass a number.

Reply:
done